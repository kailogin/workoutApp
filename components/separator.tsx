import React from "react";
import { StyleSheet, View } from "react-native";

import { Colors } from "../utils/theme";

interface SeparatorProps {
  widthPercentage?: number;
}

export const Separator = ({ widthPercentage: width }: SeparatorProps) => (
  <View style={[styles.separator, { width: `${width}%` }]} />
);

const styles = StyleSheet.create({
  separator: {
    backgroundColor: Colors.WHITE,
    height: 1,
    marginBottom: 32,
    width: "80%",
  },
});
