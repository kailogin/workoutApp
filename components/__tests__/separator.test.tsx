import "react-native";
import React from "react";
import renderer from "react-test-renderer";

import { Separator } from "../separator";

test(`BaseText snapshot`, () => {
  const snap = renderer.create(<Separator />).toJSON();

  expect(snap).toMatchSnapshot();
});
