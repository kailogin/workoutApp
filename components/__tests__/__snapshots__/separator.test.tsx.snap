// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseText snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "height": 1,
        "marginBottom": 32,
        "width": "80%",
      },
      Object {
        "width": "undefined%",
      },
    ]
  }
/>
`;
